src/controllers/expensesController.js

```diff
@@ -0,0 +1,188 @@
+'use strict';
+
+const { validate } = require('uuid');
+const expensesService = require('./../services/expensesService');
+const usersService = require('./../services/usersService');
+
+const get = async(req, res) => {
+  let expenses;
+
+  if (req.query.from && req.query.to) {
+    if (!(Date.parse(req.query.from) > 0)
+      || !(Date.parse(req.query.to) > 0)
+      || Object.keys(req.query).length > 2) {
+      res.sendStatus(400);
+    }
+
+    try {
+      expenses = await expensesService
+        .getByDate(req.query.from, req.query.to);
+
+      res.send(expenses);
+    } catch (error) {
+      res.sendStatus(500);
+    }
+
+    return;
+  }
+
+  if (req.query.userId && req.query.categories) {
+    if (!validate(req.query.userId) || Object.keys(req.query).length > 2) {
+      res.sendStatus(400);
+    }
+
+    try {
+      expenses = await expensesService
+        .getByCategory(req.query.userId, req.query.categories);
+
+      res.send(expenses);
+    } catch (error) {
+      res.sendStatus(500);
+    }
+
+    return;
+  }
+
+  if (req.query.userId) {
+    if (!validate(req.query.userId) || Object.keys(req.query).length > 1) {
+      res.sendStatus(404);
+    }
+
+    try {
+      expenses = await expensesService.getByUserId(req.query.userId);
+
+      res.send(expenses);
+    } catch (error) {
+      res.sendStatus(500);
+    }
+
+    return;
+  }
+
+  if (!Object.keys(req.query).length) {
+    try {
+      expenses = await expensesService.getAll();
+
+      res.send(expenses);
+    } catch (error) {
+      res.sendStatus(500);
+    }
+  } else {
+    res.sendStatus(400);
+  }
+};
```

Comment:
to much logic inside controller, it's not a correct approach. you should move this to service, because the main task of controller is to take request and send response. All business logic should be inside service

Reply:
https://en.wikipedia.org/wiki/Model%E2%80%93view%E2%80%93controller
The controller responds to the user input and performs interactions on the data model objects. The controller receives the input, optionally validates it and then passes the input to the model.

Reply:
@vlkzmn the previous mentor is right, controllers should have as little logic as possible, they are just middleware between route and service, so pls consider moving the majority of logic to the services
[see the article](https://www.coreycleary.me/what-is-the-difference-between-controllers-and-services-in-node-rest-apis)